### ISSUE TITLE
Batman: Arkham Asylum demo slow frame rate in CSMT

### ISSUE THREAD

**Issue by Eignar17** [wine-compholio/wine-staging#256]
Is bug https://bugs.winehq.org/show_bug.cgi?id=23993 is bat bat frame rate in csmt and wine-d3d is lag in scene

**Comment by Eignar17** [comment]
Is bug Similar.https://bugs.winehq.org/show_bug.cgi?id=24471

**Comment by slackner** [comment]
I don't think that both issues are similar. The first one is about bad performance, the second one about occasional crashes.

Based on the Appdb ratings (see https://appdb.winehq.org/objectManager.php?sClass=version&iId=19065) it seems to work for most people, which makes me think that those bugs are just outdated. The first bug is from about 3 years ago, the second one about 4 years, without any recent updates. Wine has improved a lot since then, and we would at first need some more recent feedback before we can start investigating.

I am also not sure exactly what you mean. Did you already test with Wine-Staging and/or CSMT? If those issues are still present then update the corresponding upstream bugreports.

I would recommend to test with both vanilla Wine 1.7.34 and Wine-Staging 1.7.34. Since the game can use PhysX, and we have hardware PhysX support in 1.7.34, I would expect that it runs much better with that (at least if you have an NVIDIA graphic card).

**Comment by felixonmars** [comment]
I have tried to play the game for ~0.5 hour while adjusting the settings. I have a 650Ti and I am getting similar fps when disabling PhysX or enabling PhysX with the latest PhysX hardware acceleration support in wine-staging. There were a few small lags (200~300ms) but the overall performance is acceptable (~60fps).

**Comment by Eignar17** [comment]
is Wine-Staging 1.7.34 is normal in csmt is problem a several games I will try with the new wine I notice :)

**Comment by deri82** [comment]
It's not just the wine that has changes, also people upgrade hardware, Also drivers have changed a lot meanwhile. 

I could test this but I have so powerfull gpu that it wouldnt make much sense.

**Comment by Eignar17** [comment]
yes update drivers nvidia and new wine-Staging is my nvidia geforce 210 is hardware is slow? a other games work well with csmt

**Comment by DarkPlayer** [comment]
@Eignar17 Your card is no longer supported by nvidia. The most current beta driver already dropped support for all cards below Geforce 300. You will not be able use so any driver newer than version 340.

**Comment by Eignar17** [comment]
ok thanks

**Comment by Eignar17** [comment]
Is bug in csmt

**Comment by tchebb** [comment]
I believe I'm seeing this bug as well. When CSMT is enabled, Arkham Asylum occasionally drops to unplayable framerates (between 0.5fps and 3fps, by my estimate). There doesn't seem to be an obvious event that triggers the frame drops: I've seen them occur on the main menu, in (live rendered) cutscenes, and in-game. Sometimes, the framerate goes back up on its own after 5 or 6 seconds, but most of the time I need to quit and restart the game.

The most reproducible occurrence of the issue I've found is in the Botanical Gardens, near the end of the game (I believe it's when you're on your way to the final battle with Poison Ivy): the frame rate is perfectly fine until you enter a particular room, at which point it drops. Looking at the floor or the ceiling brings it back up to normal, and looking at the walls again makes it drop. I didn't notice anything unusual about the wall geometry that could cause it, unfortunately. Once I restarted the game with CSMT disabled, the frame rate was stable throughout the entire level (and the rest of the game, for that matter).

PhysX doesn't seem to have any effect on the bug; I see the same symptoms with NVAPI disabled. The game runs fine with PhysX enabled and CSMT disabled, although I didn't notice any significant performance gain from PhysX.

All of these results are from wine-staging 1.7.37 running on Arch Linux (kernel 3.18.6), with an NVIDIA GTX 760 graphics card and the proprietary drivers. I do have CUDA properly set up, and I've verified that PhysX demos work within Wine, if that matters.

Is there any debug information I can provide to possibly help narrow down the cause? I'm afraid I'm not all that familiar with graphics programming, but I could at least help figure out where the issue is.

**Comment by DarkPlayer** [comment]
>PhysX doesn't seem to have any effect on the bug; I see the same symptoms with NVAPI disabled. The game runs fine with PhysX enabled and CSMT disabled, although I didn't notice any significant performance gain from PhysX.

Most games disable PhsyX if GPU support is unavailable or reduce the effects a lot, so it is not surprising that you don't see a performance gain.

I fear that @stefand needs to investigate this bug, but he didn't work very much on the CSMT lately.

**Comment by deri82** [comment]
How do you install/run this correctly? I have this game locally now. It installs fine. But when I launch it via launcher I cant see a thing. See more details

https://bugs.winehq.org/show_bug.cgi?id=38147

Pressing play doesnt do a thing.

**Comment by tchebb** [comment]
I've just been running it through Steam. I have `dwrite` disabled for Steam, which fixes an invisible text issue there. I think I also have `dotnet20` installed.

**Comment by deri82** [comment]
I dont remember having this issue with the full game (steam too).
I have it installed on crossover.

**Comment by slackner** [comment]
Hi everyone,

About one year ago, we announced that we would like to move to more advanced bug tracker. We are sorry for the inconvenience, but now its time to say finally goodbye to the old one. This means, all remaining opened bugs (including this one!) will be closed, and no longer available. If you still need any information from the old bugtracker, please write a mail to webmaster@fds-team.de.

More important, if your issue still exists, please open a new bug report at one of the places mentioned below. For bug reports, please attach terminal output from a recent Wine version (preferably >= 1.8). Please note that in the meantime, Wine Staging has become an official part of WineHQ, so if your issue is already tracked at winehq.org, there is NO NEED to open a new bug report for it. Feel free to add the Wine Staging maintainers (sebastian@fds-team.de, michael@fds-team.de) as CC if you want us to take a look at your bug report.

**NEW BUG TRACKERS:**

* For all Wine AND Wine Staging bugs. USE THIS IF IN DOUBT. (*)
  https://bugs.winehq.org/enter_bug.cgi?product=Wine

* For packaging issues:
  https://bugs.winehq.org/enter_bug.cgi?product=Packaging

* For patch submission:
  https://bugs.wine-staging.com/enter_bug.cgi

Best regards,
Sebastian

----

(*) The "Wine Staging" product in the winehq bugtracker should only be used, when your problem turns out to be a regression introduced in "Wine Staging". Since most problems also occur with Wine, it is recommended to open your bug in the "Wine" product, unless someone can show that it doesn't occur with the development branch.
